### DISCUSSION TITLE
Interpolated string - Do string literals cause a performance overhead?

### DISCUSSION THREAD

**Discussion by 0x90d** [dotnet/runtime#59016] [Q&A] [answered]
To improve readability the MSDN tells me that I can use string literals and this will be [concated at compile time](https://docs.microsoft.com/en-us/dotnet/csharp/how-to/concatenate-multiple-strings#string-literals) so there is no performance penalty. My question is, is this also true for interpolated strings? For example:

```C#
string text = $"{a} Historically, the world of data and the world of objects " +
$"{b} have not been well integrated. Programmers work in C# or Visual Basic " +
$"{c} and also in SQL or XQuery. On the one side are concepts such as classes, " +
$"{d} objects, fields, inheritance, and .NET Framework APIs. On the other side " +
$"{e} are tables, columns, rows, nodes, and separate languages for dealing with ";
```
Will this get compiled the same as:
```C#
string text = $"{a} Historically, the world of data and the world of objects {b} have not been well integrated. Programmers work in C# or Visual Basic {c} and also in SQL or XQuery. On the one side are concepts such as classes, {d} objects, fields, inheritance, and .NET Framework APIs. On the other side {e} are tables, columns, rows, nodes, and separate languages for dealing with ";
```
And if not, which one performs better?

**Comment by ufcpp** [comment]
https://github.com/dotnet/roslyn/pull/26612

**Reply by 0x90d** [reply]
Can you please write a summary of what the linked issue is supposed to tell me? I don't want to read hundred of posts in order to receive a a simple answer of my simple question. Thank you

**Comment by stephentoub** [comment, marked as answer]
> Will this get compiled the same as

Yes
https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6zT1sAGEgWWwApm2I4yrEsAG8SAY1FCEUtgFMAlCQC8APjoNNAEgBE1CAF8SACSgBnAC4B7AE5RxEADaOAngIsALOSQDutxwgkVgBmJAgQFhAkEDCBnt5+NgFBoVbAAFZy4hZmJDokyBqa9LrUwEYeEABu3jBWFsJycjC+cs4ksBZyAOY2EXIIAHSCAOxgcja5iQDWHS0UAMRBNiQAauYArk4kAEIQZvZ5BUXFpeJGMYFOZlZzJADKAIoAMsskABqPGxMuwwDyLXiQRg3gOCG8EBs3nEVhg4jkAAcciQzBtxB5orksHIzAJ8oV8MUGKUEEY0plsriSMEoG0EFTYF47JE4XIBJcSIMAHIAUQAKiQAGJ9cYzEgAQQACgBJMz/QFeILxFZg7z4k7aPRyC5QkiRYCOHECGGODZgGBUmxWHxUurgqkcsyIyH9EiOGLdLbdHHU2xhOROWDdXxQTx5ADcJxOxEEbHYmB4Qn4seEsgkaZkKcUKnUhKJJFKhhM5msdgczjcesViWSITCESiHKBNcCdfJWWRZQq1Vq9UazVa7U6PT6XSGo3Gk18tlmsEoSz961R2z2B3ENHO0Vi0UcNzuT1efs+3xsvxIAKr3lhoKg4OiuphrKRuVR6MxZEwRpopKCGQ7VJpOkGRgJlQxieF2W3bl+SFEU5DFKVZXlS8lSZFFb28ahtXvbx9UNKkTTNC0BCtG0BDtL9HWdUdvHdGBPQgb1cmCP1wUDeiQzDHRI3wAwgA===

**Reply by 0x90d** [reply]
@stephentoub Thank you
